tomcat-gcp-lib/src/main/java/com/google/cloud/runtimes/tomcat/logging/TraceValve.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.tomcat.logging;
+
+import com.google.cloud.trace.Trace;
+import com.google.cloud.trace.Tracer;
+import com.google.cloud.trace.core.SpanContext;
+import com.google.cloud.trace.core.TraceContext;
+import com.google.cloud.trace.service.TraceGrpcApiService;
+import com.google.cloud.trace.service.TraceService;
+
+import org.apache.catalina.LifecycleException;
+import org.apache.catalina.connector.Request;
+import org.apache.catalina.connector.Response;
+import org.apache.catalina.valves.ValveBase;
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.ServletException;
+
+/**
+ * This valve sends information about the requests to the Stackdriver Trace service.
+ */
+public class TraceValve extends ValveBase {
+
+  /**
+   * Header used by GCP to stores the trace id and the span id.
+   */
+  private static final String X_CLOUD_TRACE_HEADER = "x-cloud-trace-context";
+
+  private static final Log log = LogFactory.getLog(TraceValve.class);
+
+  private TraceService traceService;
+
+  /**
+   * Delay in second before the trace scheduler send the traces (allow buffering or traces).
+   */
+  private Integer traceDelay;
```

Comment:
I would rename this to ``traceInterval`` - as ``delay`` is more of a "one time" scheduling concept, ``interval`` is closer to the meaning of this variable that on every x seconds the buffered traces are reported.

We can also go with ``traceScheduledDelay`` to stay consistent with the Trace gRPC API.